ClientMultiMapLockTest.testLockTTLExpired_whenLockedBySelf

```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at org.junit.Assert.assertFalse(Assert.java:74)
	at com.hazelcast.client.multimap.ClientMultiMapLockTest.testLockTTLExpired_whenLockedBySelf(ClientMultiMapLockTest.java:328)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-OracleJDK1.6/com.hazelcast$hazelcast-client-legacy/740/testReport/junit/com.hazelcast.client.multimap/ClientMultiMapLockTest/testLockTTLExpired_whenLockedBySelf/

--- a/hazelcast-client-legacy/src/test/java/com/hazelcast/client/multimap/ClientMultiMapLockTest.java
+++ b/hazelcast-client-legacy/src/test/java/com/hazelcast/client/multimap/ClientMultiMapLockTest.java
@@ -313,8 +313,12 @@ public class ClientMultiMapLockTest extends HazelcastTestSupport {
         final Object key = "Key";
 
         mm.lock(key, 1, TimeUnit.SECONDS);
-        sleepSeconds(2);
-        assertFalse(mm.isLocked(key));
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                assertFalse(mm.isLocked(key));
+            }
+        });
     }
 
     @Test

--- a/hazelcast-client-legacy/src/test/java/com/hazelcast/client/multimap/ClientMultiMapLockTest.java
+++ b/hazelcast-client-legacy/src/test/java/com/hazelcast/client/multimap/ClientMultiMapLockTest.java
@@ -324,8 +324,13 @@ public class ClientMultiMapLockTest extends HazelcastTestSupport {
 
         mm.lock(key);
         mm.lock(key, 1, TimeUnit.SECONDS);
-        sleepSeconds(2);
-        assertFalse(mm.isLocked(key));
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                assertFalse(mm.isLocked(key));
+            }
+        });
     }
 
     @Test

--- a/hazelcast-client/src/test/java/com/hazelcast/client/multimap/ClientMultiMapLockTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/multimap/ClientMultiMapLockTest.java
@@ -313,8 +313,12 @@ public class ClientMultiMapLockTest extends HazelcastTestSupport {
         final Object key = "Key";
 
         mm.lock(key, 1, TimeUnit.SECONDS);
-        sleepSeconds(2);
-        assertFalse(mm.isLocked(key));
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                assertFalse(mm.isLocked(key));
+            }
+        });
     }
 
     @Test
@@ -324,8 +328,13 @@ public class ClientMultiMapLockTest extends HazelcastTestSupport {
 
         mm.lock(key);
         mm.lock(key, 1, TimeUnit.SECONDS);
-        sleepSeconds(2);
-        assertFalse(mm.isLocked(key));
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                assertFalse(mm.isLocked(key));
+            }
+        });
+
     }
 
     @Test

